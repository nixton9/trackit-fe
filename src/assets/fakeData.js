export const notes = [
  {
    id: 1,
    title: 'Lista de compras',
    date: '2020-07-20',
    content: 'asdasdad',
    tags: [
      { id: 1, name: 'Primary', color: '#F44336' },
      { id: 2, name: 'Secondary', color: '#E91E63' }
    ]
  },
  {
    id: 2,
    title: 'Apontamentos aula',
    date: '2020-06-25',
    content: 'asdasdad',
    tags: [
      { id: 3, name: 'Studies', color: '#607D8B' },
      { id: 4, name: 'School', color: '#00BCD4' }
    ]
  },
  {
    id: 3,
    title: 'Random shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [{ id: 5, name: 'Random', color: '#3F51B5' }]
  },
  {
    id: 4,
    title: 'Work shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [
      { id: 6, name: 'Work', color: '#9C27B0' },
      { id: 7, name: 'Coding', color: '#FF9800' },
      { id: 8, name: 'Stuff', color: '#4CAF50' }
    ]
  },
  {
    id: 5,
    title: 'Post 129',
    date: '2020-05-07',
    content: 'asdasdad'
  }
]
