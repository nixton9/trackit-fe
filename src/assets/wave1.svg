<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_294" data-name="Caminho 294" d="M-14697.143-11591.146s204.548,93.02,241.44,226.007,101.77,246.085,225.016,271.63,239.81-133.94,286.815-109.888,72.048,136.969,72.048,136.969,18.206,112.27,0,146.063-154.687,113.568-154.687,113.568-145.483,41.1-252.11,41.649-282.684-4.542-332.2-22.729-57.569-16.607-86.319-46.7-74.028-81.864-81.211-123.045-11.661-111.631-8.345-135.894,30.468-240.38,30.94-257.871-.572-130.1,0-136.284S-14697.143-11591.146-14697.143-11591.146Z" transform="translate(14695 11649)" fill="#273150" opacity="0.051"/>
  </g>
</svg>
