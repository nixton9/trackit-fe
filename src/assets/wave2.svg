<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M-14243.049-11762.149s99.251-140.3,221.355-63.8,71.256,212.9,188.94,256.211,284.029-180.642,371.026-162.149,131.877,179.9,131.877,179.9L-13407-11443.72l-79.384,66.653s-188.631,35.67-195.492,37.041-321.083,20.295-351.518,20.295-209.207-15.854-221.288-20.295-32.708-292.639-32.708-292.639Z" transform="translate(14215 12160.858)" fill="#273150" opacity="0.041"/>
  </g>
</svg>
