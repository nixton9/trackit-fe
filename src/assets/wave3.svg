<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_308" data-name="Caminho 308" d="M1430.13,18293.221s356.522-204.141,537.172-65.336,273.16-352.586,456.213-279.107,399.449,358.18,399.449,358.18,205.3,357.1,208.9,365.451-160.994,153.961-208.9,147.217-546.637-79.807-705.6-97.068-577.556-19.437-656.578-28.982-318.938-34.217-335.6-53.873,280.937-332.744,280.937-332.744Z" transform="translate(-1614 -17637)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
