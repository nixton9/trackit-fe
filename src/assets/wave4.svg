<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M1449.88,18572s112.775-319.779,271.458-247.187,168.861,179.166,303.627,151.676,170.134-339.822,339.459-310.615,218.6,83.457,218.6,83.457l52.227,227.158v247.463s-162.445,46.264-172.641,44.359-378.379,27.41-395.417,29.01-439.332,0-439.332,0l-251.667-29.01Z" transform="translate(-1613 -17892)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
