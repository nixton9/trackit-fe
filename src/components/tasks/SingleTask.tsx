import React, { useState } from 'react'
import { Styled } from '../../styles/Single.styles'
import { Task } from '../../utils/ModuleTypes'
import {
  displayDateString,
  isPastDate,
  parseDateInverse
} from '../../utils/dateHelpers'
import { TASKS } from '../../utils/queries'
import { ReactComponent as CalendarIcon } from '../../assets/icons/calendr.svg'
import { gql, useMutation, useQuery } from '@apollo/client'

const MARK_AS_DONE = gql`
  mutation MarkAsDone($id: ID!, $done: Boolean) {
    updateTask(id: $id, done: $done) {
      id_task
      done
    }
  }
`

const SingleTask: React.FC<Task> = ({ id, title, date, done, category }) => {
  const [taskDone, setTaskDone] = useState(false)

  const { refetch: refetchTasks } = useQuery(TASKS)
  const [markAsDone] = useMutation(MARK_AS_DONE, {
    variables: { id: id, done: true }
  })

  const handleCompleteTask = (e: React.MouseEvent<HTMLDivElement>) => {
    setTaskDone(true)
    markAsDone()
      .then(res => refetchTasks())
      .catch(err => console.log(err))
  }

  return (
    <Styled.SingleContainer>
      <Styled.SingleFlex>
        <div>
          <Styled.SingleFlex>
            <Styled.SingleTitle>{title}</Styled.SingleTitle>
            {category && (
              <Styled.SingleCategory color={category.color}>
                {category.name}
              </Styled.SingleCategory>
            )}
          </Styled.SingleFlex>
          <Styled.SingleDate past={isPastDate(parseDateInverse(date))}>
            <CalendarIcon />
            <p>{displayDateString(parseDateInverse(date))}</p>
          </Styled.SingleDate>
        </div>
        <Styled.SingleTask__Complete
          onClick={handleCompleteTask}
          className={taskDone ? 'done' : ''}
        />
      </Styled.SingleFlex>
    </Styled.SingleContainer>
  )
}

export default SingleTask
