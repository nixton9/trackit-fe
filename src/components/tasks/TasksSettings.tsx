import React, { useState, useEffect, Dispatch, SetStateAction } from 'react'
import Drawer from '../misc/Drawer'
import { CategoryEditor } from './CategoryEditor'
import { SelectMenu } from '../misc/SelectMenu'
import { NotificationTypes, notificationState } from '../misc/Notification'
import { alertState } from '../misc/Alert'
import { Styled } from '../../styles/Settings.styles'
import { useToggleElement } from '../../utils/useToggleElement'
import { CATEGORIES, TASKS } from '../../utils/queries'
import { ReactComponent as SettingsIcon } from '../../assets/icons/settings.svg'
import { ReactComponent as PlusIcon } from '../../assets/icons/plus.svg'
import { ReactComponent as MinusIcon } from '../../assets/icons/minus.svg'
import { TaskCategory } from '../../utils/ModuleTypes'
import { SortBySettings, sortByOptions } from '../../utils/SettingsTypes'
import { useSetRecoilState } from 'recoil'
import { gql, useMutation, useQuery } from '@apollo/client'
import { useHistory } from 'react-router-dom'

const DELETE_CATEGORY = gql`
  mutation DeleteCategory($id: ID!) {
    deleteCategory(id: $id) {
      id_category
    }
  }
`

type TasksSettingsProps = {
  categories: TaskCategory[]
  sortBy: string
  setSortBy: Dispatch<SetStateAction<SortBySettings>>
}

const TasksSettings: React.FC<TasksSettingsProps> = ({
  categories,
  sortBy,
  setSortBy
}) => {
  const [showCategoryEditor, setShowCategoryEditor] = useState(false)
  const [activeCategory, setActiveCategory] = useState<TaskCategory | null>(
    null
  )

  const setNotification = useSetRecoilState(notificationState)
  const setAlert = useSetRecoilState(alertState)

  const history = useHistory()

  const [open, setOpen, overlayEl] = useToggleElement(() =>
    setShowCategoryEditor(false)
  )

  const { refetch: refetchCategories } = useQuery(CATEGORIES, {
    fetchPolicy: 'network-only'
  })
  const { refetch: refetchTasks } = useQuery(TASKS)

  const [deleteCategory] = useMutation(DELETE_CATEGORY)

  const handleSortByChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    setSortBy(e.target.value as SortBySettings)

  const handleCategoryClick = (category: TaskCategory) => {
    setActiveCategory(category)
    setShowCategoryEditor(true)
  }

  const handlePlusClick = () => {
    if (showCategoryEditor) {
      setShowCategoryEditor(false)
    } else {
      setActiveCategory(null)
      setShowCategoryEditor(true)
    }
  }

  const handleDeleteType = (
    categoryId: string | number,
    categoryName: string
  ) => {
    deleteCategory({
      variables: {
        id: categoryId
      }
    })
      .then(res => {
        setNotification({
          text: `Inbox '${categoryName}' was deleted`,
          type: NotificationTypes.Success
        })
        refetchCategories()
        refetchTasks()
      })
      .catch(err =>
        setNotification({
          text: 'There was a problem, please try again',
          type: NotificationTypes.Error
        })
      )
  }

  const handleDeleteTypeConfirm = (
    categoryId: string | number,
    categoryName: string
  ) => {
    setShowCategoryEditor(false)
    setAlert({
      text: 'This inbox will be removed',
      onConfirm: () => handleDeleteType(categoryId, categoryName)
    })
  }

  useEffect(() => {
    history.push(`#`)

    window.onpopstate = (e: any) => {
      setOpen(false)
    }
  }, [history, setOpen])

  return (
    <>
      <SettingsIcon
        className="settings-icon"
        onClick={() => setOpen(true)}
        data-test-id="tasks-settings-icon"
      />

      <Drawer
        title="Settings"
        open={open}
        setOpen={setOpen}
        overlayRef={overlayEl}
      >
        <>
          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label htmlFor="sort-by">
              Sort by
            </Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Input>
              <SelectMenu
                id="sort-by"
                value={sortBy}
                onChange={handleSortByChange}
                options={sortByOptions}
              />
            </Styled.SettingsBlock__Input>
          </Styled.SettingsBlock>

          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label>Inboxes</Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Categories>
              {categories &&
                categories.map(cat => (
                  <Styled.SettingsBlock__Category
                    color={cat.color}
                    key={cat.id}
                  >
                    <div
                      className="inner"
                      onClick={() => handleCategoryClick(cat)}
                    >
                      {cat.name}
                    </div>
                    <span
                      className="delete"
                      onClick={() => handleDeleteTypeConfirm(cat.id, cat.name)}
                    >
                      +
                    </span>
                  </Styled.SettingsBlock__Category>
                ))}
              <Styled.SettingsBlock__Icon
                className="mbl-click"
                onClick={handlePlusClick}
                data-test-id="categories-add-icon"
              >
                {showCategoryEditor ? <MinusIcon /> : <PlusIcon />}
              </Styled.SettingsBlock__Icon>
            </Styled.SettingsBlock__Categories>
          </Styled.SettingsBlock>

          {showCategoryEditor && (
            <CategoryEditor
              category={activeCategory}
              closeEditor={() => setShowCategoryEditor(false)}
            />
          )}
        </>
      </Drawer>
    </>
  )
}

export default TasksSettings
