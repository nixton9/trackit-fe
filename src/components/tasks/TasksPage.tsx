import React, { useState } from 'react'
import TasksSettings from './TasksSettings'
import SingleTask from './SingleTask'
import Tooltip from 'react-tooltip-lite'
import { SelectMenu } from '../misc/SelectMenu'
import { PageLoading } from '../misc/PageLoading'
import { PageError } from '../misc/PageError'
import { activeContentState } from '../misc/Add'
import { Styled } from '../../styles/Page.styles'
import { Task, ModuleTypes } from '../../utils/ModuleTypes'
import { SortBySettings } from '../../utils/SettingsTypes'
import { TASKS, CATEGORIES } from '../../utils/queries'
import { isDateToday, parseDateInverse } from '../../utils/dateHelpers'
import { tasksViewOptions } from '../../utils/selectsOptions'
import { sortData } from '../../utils/globalHelpers'
import { ReactComponent as PlusIcon } from '../../assets/icons/plus.svg'
import { ReactComponent as NoDataIcon } from '../../assets/icons/nodata.svg'
import { ReactComponent as TasksIcon } from '../../assets/icons/tasks.svg'
import { ReactComponent as DoneTasksIcon } from '../../assets/icons/donetasks.svg'
import { useQuery } from '@apollo/client'
import { useSetRecoilState } from 'recoil'
import { Link } from 'react-router-dom'

type TasksPageProps = {
  done?: boolean
}

const TasksPage: React.FC<TasksPageProps> = ({ done }) => {
  const setActiveContent = useSetRecoilState(activeContentState)

  const { loading, error, data } = useQuery(TASKS, {
    variables: { done: done }
  })
  const { data: categories } = useQuery(CATEGORIES)

  const [view, setView] = useState('all')
  const [sortBy, setSortBy] = useState<SortBySettings>(SortBySettings.DATE)

  const handleViewChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    setView(e.target.value)

  const visibleTasks = data
    ? view === 'all'
      ? data.tasks
      : view === 'today'
      ? data.tasks.filter((task: Task) =>
          task.date ? isDateToday(parseDateInverse(task.date)) : false
        )
      : data.tasks.filter(
          (task: Task) => Number(task.category?.id) === Number(view)
        )
    : []

  const sortedTasks = sortData(visibleTasks, sortBy, true)

  return (
    <Styled.PageContainer className="overflow">
      <Styled.PageTitle>{done && 'Completed'} Tasks</Styled.PageTitle>

      <Styled.PageHeader className="page-header">
        <Styled.PageHeader__View>
          <Styled.PageHeader__View__Dropdown className="tasks">
            <SelectMenu
              id="tasks-view"
              value={view}
              onChange={handleViewChange}
              options={tasksViewOptions(categories)}
              itemClass={'view-select-item'}
            />
          </Styled.PageHeader__View__Dropdown>

          <Tooltip
            tipContentClassName="visible-tooltip"
            content={`${visibleTasks.length} tasks in view`}
            arrow={false}
            direction={'up'}
          >
            <Styled.PageHeader__View__Counter>
              {visibleTasks.length}
            </Styled.PageHeader__View__Counter>
          </Tooltip>
        </Styled.PageHeader__View>

        <Styled.PageHeader__Settings>
          {done ? (
            <Tooltip
              eventOff={'onClick'}
              content={'Tasks'}
              arrow={false}
              direction={'up'}
            >
              <Link to="/tasks" className="mbl-click">
                <TasksIcon />
              </Link>
            </Tooltip>
          ) : (
            <Tooltip
              eventOff={'onClick'}
              content={'Done Tasks'}
              arrow={false}
              direction={'up'}
            >
              <Link
                to="/tasks/done"
                className="mbl-click"
                data-test-id="done-tasks-link"
              >
                <DoneTasksIcon />
              </Link>
            </Tooltip>
          )}

          <Tooltip
            eventOff={'onClick'}
            content={'Settings'}
            arrow={false}
            direction={'up'}
          >
            <div className="mbl-click">
              <TasksSettings
                categories={categories ? categories.categories : []}
                sortBy={sortBy}
                setSortBy={setSortBy}
              />
            </div>
          </Tooltip>
        </Styled.PageHeader__Settings>
      </Styled.PageHeader>

      <Styled.PageContent className="overflow">
        {error ? (
          <PageError>Couldn't get data, check your connection.</PageError>
        ) : loading ? (
          <PageLoading />
        ) : sortedTasks.length ? (
          (sortedTasks as Task[]).map(task => (
            <SingleTask
              key={task.id}
              id={task.id}
              title={task.title}
              date={task.date}
              done={task.done}
              category={task.category}
              disableStatus={done}
            />
          ))
        ) : (
          <Styled.PageContent__NoData>
            <NoDataIcon />
          </Styled.PageContent__NoData>
        )}
      </Styled.PageContent>
      <Styled.PageAddItem
        onClick={() => setActiveContent(ModuleTypes.Tasks)}
        data-test-id="add-task"
      >
        <PlusIcon />
      </Styled.PageAddItem>
    </Styled.PageContainer>
  )
}

export default TasksPage
