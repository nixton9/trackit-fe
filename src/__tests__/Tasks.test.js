import React from 'react'
import { render } from '@testing-library/react'
import TasksPage from '../components/tasks/TasksPage'
import SingleTask from '../components/tasks/SingleTask'
import TasksSettings from '../components/tasks/TasksSettings'
import { tasks, tasksCategories } from '../assets/fakeData'
import { BrowserRouter } from 'react-router-dom'
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: '/',
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjYsImlhdCI6MTU5NDU3Nzc1M30.BLYTV2jBpTk3PyMU7j9-53FAXEzAY6KuBH79mIbZvho`
  }
})

describe('Tasks Page', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <TasksPage />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has the title', () => {
    const { getByText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <TasksPage />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByText('Tasks')).toBeInTheDocument()
  })
})

describe('SingleTask', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SingleTask
            id={tasks[0].id}
            title={tasks[0].title}
            date={tasks[0].date}
            done={tasks[0].done}
            category={tasks[0].category}
          />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the content of the task', () => {
    const { getByText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SingleTask
            id={tasks[0].id}
            title={tasks[0].title}
            date={tasks[0].date}
            done={tasks[0].done}
            category={tasks[0].category}
          />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByText('Ir Ã s compras')).toBeInTheDocument()
    expect(getByText('14 Aug')).toBeInTheDocument()
    expect(getByText('Waiting for')).toBeInTheDocument()
  })
})

describe('TasksSettings', () => {
  it('matches snashot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <TasksSettings tags={tasksCategories} />
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the fields', () => {
    const { getByText } = render(
      <ApolloProvider client={client}>
        <TasksSettings tags={tasksCategories} />
      </ApolloProvider>
    )

    expect(getByText('Sort by')).toBeInTheDocument()
    expect(getByText('Font size')).toBeInTheDocument()
    expect(getByText('Categories')).toBeInTheDocument()
  })

  it('shows tags', () => {
    const { getByText } = render(
      <ApolloProvider client={client}>
        <TasksSettings categories={tasksCategories} />
      </ApolloProvider>
    )

    expect(getByText('Waiting for')).toBeInTheDocument()
    expect(getByText('House')).toBeInTheDocument()
    expect(getByText('Trackit')).toBeInTheDocument()
    expect(getByText('Stuff')).toBeInTheDocument()
  })
})
