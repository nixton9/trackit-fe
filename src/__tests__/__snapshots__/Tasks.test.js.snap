// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTask matches snapshot 1`] = `
<div>
  <article
    class="sc-fzpans gHCdfm"
  >
    <div
      class="sc-fzplWN gwxuzL"
    >
      <div>
        <div
          class="sc-fzplWN gwxuzL"
        >
          <h5
            class="sc-fznyAO drGMcz"
          >
            Ir às compras
          </h5>
          <h6
            class="sc-fznZeY cjOYaZ"
            color="#673AB7"
          >
            Waiting for
          </h6>
        </div>
        <div
          class="sc-fznKkj jNhHYx"
        >
          <svg>
            calendr.svg
          </svg>
          <p>
            14 Aug
          </p>
        </div>
      </div>
      <div
        class="sc-fzoyAV bnnRRV"
      />
    </div>
  </article>
</div>
`;

exports[`Tasks Page matches snapshot 1`] = `
<div>
  <div
    class="sc-fzpjYC izvAVX"
  >
    <h1
      class="sc-fznxsB iTQhax"
    >
      Tasks
    </h1>
    <div
      class="sc-fznJRM cBJzUp"
    >
      <div
        class="sc-fznWqX dZCTFs"
      >
        <div
          class="sc-fzoiQi gKygiC"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="tasks-view"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="tasks-view"
              role="button"
              tabindex="0"
            >
              Today
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="today"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
        <h3
          class="sc-fzqARJ lfDBcG"
        >
          0
        </h3>
      </div>
      <div
        class="sc-fzqNqU hQTkrp"
      >
        <svg
          class="settings-icon"
        >
          settings.svg
        </svg>
        <div
          class="sc-AxjAm khxSQh"
        >
          <h2
            class="sc-AxiKw jzoYkp"
          >
            Settings
          </h2>
          <div
            class="sc-AxhCb bCDkJU"
          >
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
                for="sort-by"
              >
                Sort by
              </label>
              <div
                class="sc-AxheI hCWMTo"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="sort-by"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="sort-by"
                    role="button"
                    tabindex="0"
                  >
                    Date
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="DATE"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
                for="font-size"
              >
                Font size
              </label>
              <div
                class="sc-AxheI hCWMTo"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="font-size"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="font-size"
                    role="button"
                    tabindex="0"
                  >
                    Medium
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="medium"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
              >
                Categories
              </label>
              <div
                class="sc-AxmLO ceYEas"
              >
                <span
                  class="sc-fzoLsD bpeBKU"
                >
                  <svg>
                    plus.svg
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxirZ gpexXZ"
        />
      </div>
    </div>
    <section
      class="sc-fzoyTs gUbeoJ"
    >
      <div
        class="sc-fznMAR gCPQx"
      >
        <div
          class="sc-fzoLag kUuTIm"
        >
          <div
            class="sc-fzoXzr hScLKH"
          />
          <div
            class="sc-fzoXzr hScLKH doublebounce2"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`TasksSettings matches snashot 1`] = `
<div>
  <svg
    class="settings-icon"
  >
    settings.svg
  </svg>
  <div
    class="sc-AxjAm khxSQh"
  >
    <h2
      class="sc-AxiKw jzoYkp"
    >
      Settings
    </h2>
    <div
      class="sc-AxhCb bCDkJU"
    >
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
          for="sort-by"
        >
          Sort by
        </label>
        <div
          class="sc-AxheI hCWMTo"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="sort-by"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="sort-by"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
          for="font-size"
        >
          Font size
        </label>
        <div
          class="sc-AxheI hCWMTo"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="font-size"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="font-size"
              role="button"
              tabindex="0"
            >
              Medium
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="medium"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
        >
          Categories
        </label>
        <div
          class="sc-AxmLO ceYEas"
        >
          <span
            class="sc-fzoLsD bpeBKU"
          >
            <svg>
              plus.svg
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-AxirZ gpexXZ"
  />
</div>
`;
